dbms/src/Flash/Coprocessor/DAGStorageInterpreter.cpp

```diff
@@ -695,26 +692,18 @@ void DAGStorageInterpreter::buildRemoteExec(
     const DAGSchema & schema = remote_requests[0].schema;
     pingcap::kv::Cluster * cluster = tmt.getKVCluster();
     bool has_enforce_encode_type = remote_requests[0].dag_request.has_force_encode_type() && remote_requests[0].dag_request.force_encode_type();
-    size_t concurrent_num = std::min<size_t>(context.getSettingsRef().max_threads, all_tasks.size());
-    size_t task_per_thread = all_tasks.size() / concurrent_num;
-    size_t rest_task = all_tasks.size() % concurrent_num;
     pingcap::kv::LabelFilter tiflash_label_filter = pingcap::kv::labelFilterNoTiFlashWriteNode;
-    /// TODO: support reading data from write nodes
-    for (size_t i = 0, task_start = 0; i < concurrent_num; ++i)
-    {
-        size_t task_end = task_start + task_per_thread;
-        if (i < rest_task)
-            ++task_end;
-        if (task_end == task_start)
-            continue;
-        std::vector<pingcap::coprocessor::CopTask> tasks(all_tasks.begin() + task_start, all_tasks.begin() + task_end);
 
-        auto coprocessor_reader = std::make_shared<CoprocessorReader>(schema, cluster, tasks, has_enforce_encode_type, 1, tiflash_label_filter);
-        context.getDAGContext()->addCoprocessorReader(coprocessor_reader);
+    size_t concurrent_num = std::min<size_t>(context.getSettingsRef().max_threads, all_tasks.size());
+    size_t queue_size = context.getSettingsRef().remote_read_queue_size > 0 ? context.getSettingsRef().remote_read_queue_size.get() : concurrent_num * 4;
+    bool is_cop_stream = context.getSettingsRef().enable_cop_stream_for_remote_read;
+    auto coprocessor_reader = std::make_shared<CoprocessorReader>(schema, cluster, std::move(all_tasks), has_enforce_encode_type, concurrent_num, tiflash_label_filter, queue_size, is_cop_stream);
+    context.getDAGContext()->addCoprocessorReader(coprocessor_reader);
```

Comment:
Maybe can use a common function to create CoprocessorReader for both `buildRemoteExec` and `buildRemoteStreams`